<robot name="arm">
	<link name="world"/>
	<link name="base_link">
		<inertial>
			<mass value="1.0"/>
			<origin rpy="0 0 0" xyz="0 0 0.025"/>
			<inertia ixx="0.0027" iyy="0.0027" izz="0.005" ixy="0" ixz="0" iyz="0"/>
		</inertial>	
		<visual>
			<geometry>
				<cylinder length="0.05" radius="0.1"/>
			</geometry>
			<material name="silver">
				<color rgba="0.75 0.75 0.75 1"/>
			</material>
			<origin rpy="0 0 0" xyz="0 0 0.025"/> 
		</visual>
		<collision>
			<geometry>
				<cylinder length="0.05" radius="0.1"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0.025"/> 
		</collision>
	</link>
	<joint name="fixed" type="fixed">
		<parent link="world"/>
		<child link="base_link"/>
	</joint>
	<link name="torso">
		<inertial>
			<mass value="1.0"/>
			<origin rpy="0 0 0" xyz="0 0 0.25"/>
			<inertia ixx="0.02146" iyy="0.02146" izz="0.00125" ixy="0" ixz="0" iyz="0"/>
		</inertial>	
		<visual>		
			<geometry>
				<cylinder length="0.5" radius="0.05"/>
			</geometry>
			<material name="silver">
				<color rgba="0.75 0.75 0.75 1"/>
			</material>
			<origin rpy="0 0 0" xyz="0.0 0.0 0.25"/>
		</visual>
		<collision>
			<geometry>
				<cylinder length="0.5" radius="0.05"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0.0 0.0 0.25"/>
		</collision>
	</link>
	<joint name="hip" type="continuous">
		<axis xyz="0 0 1"/>
		<parent link="base_link"/>
		<child link="torso"/>
		<origin rpy="0 0 0" xyz="0 0 0.05"/>
	</joint>
	<link name="upper_arm">
		<inertial>
			<mass value="1.0"/>
			<origin rpy="0 0 0" xyz="0 0 0.2"/>
			<inertia ixx="0.01396" iyy="0.01396" izz="0.00125" ixy="0" ixz="0" iyz="0"/>
		</inertial>	
		<visual>
			<geometry>
				<cylinder length="0.4" radius="0.05"/>
			</geometry>
			<material name="silver"/>
			<origin rpy="0 0 0" xyz="0 0 0.2"/>
		</visual>
		<collision>
			<geometry>
				<cylinder length="0.4" radius="0.05"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0.2"/>
		</collision>
	</link>
	<joint name="shoulder" type="revolute">
		<axis xyz="0 1 0"/>
		<limit velocity="1.0" effort="30.0" upper="0.6981" lower="-1.5708"/>
		<parent link="torso"/>
		<child link="upper_arm"/>
		<origin rpy="0 1.5708 0" xyz="0.0 -0.1 0.45"/>
	</joint>
	<link name="lower_arm">
		<inertial>
			<mass value="1.0"/>
			<origin rpy="0 0 0" xyz="0 0 0.2"/>
			<inertia ixx="0.01396" iyy="0.01396" izz="0.00125" ixy="0" ixz="0" iyz="0"/>
		</inertial>
		<visual>
			<geometry>
				<cylinder length="0.4" radius="0.05"/>
			</geometry>
			<material name="silver"/>
			<origin rpy="0 0 0" xyz="0 0 0.2"/>
		</visual>
		<collision>
			<geometry>
				<cylinder length="0.4" radius="0.05"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0.2"/>
		</collision>
	</link>
	<joint name="elbow" type="revolute">
		<axis xyz="0 1 0"/>
		<limit velocity="1.0" effort="30.0" upper="1.5708" lower="-1.5708"/>
		<parent link="upper_arm"/>
		<child link="lower_arm"/>
		<origin rpy="0 0 0" xyz="0.0 0.1 0.35"/>
	</joint>
	<link name="hand">
		<inertial>
			<mass value="1.0"/>
			<inertia ixx="0.00042" iyy="0.00042" izz="0.00042" ixy="0" ixz="0" iyz="0"/>
		</inertial>
		<visual>
			<geometry>
				<box size="0.05 0.05 0.05"/>
			</geometry>
			<material name="silver"/>
		</visual>
		<collision>
			<geometry>
				<box size="0.05 0.05 0.05"/>
			</geometry>
		</collision>
	</link>
	<joint name="wrist" type="continuous">
		<axis xyz="0 1 0"/>
		<parent link="lower_arm"/>
		<child link="hand"/>
		<origin rpy="0 0 0" xyz="0.0 0.0 0.425"/>
	</joint>
	<link name="left_gripper_base">
		<inertial>
			<mass value="0.01"/>
			<inertia ixx="8.23e-8" iyy="8.23e-8" izz="3.47e-7" ixy="0" ixz="0" iyz="0"/>
		</inertial>
		<visual>
			<geometry>
				<box size="0.005 0.005 0.02"/>
			</geometry>
			<material name="silver"/>
		</visual>
		<collision>
			<geometry>
				<box size="0.005 0.005 0.02"/>
			</geometry>
		</collision>
	</link>
	<joint name="l_g_base" type="revolute">
		<axis xyz="1 0 0"/>
		<limit velocity="1.0" effort="30.0" lower="0" upper="0.5236"/>
		<origin rpy="-0.9162 0 0" xyz="0 0.02 0.035"/>
		<parent link="hand"/>
		<child link="left_gripper_base"/>
	</joint>
	<link name="right_gripper_base">
		<inertial>
			<mass value="0.01"/>
			<inertia ixx="8.23e-8" iyy="8.23e-8" izz="3.47e-7" ixy="0" ixz="0" iyz="0"/>
		</inertial>
		<visual>
			<geometry>
				<box size="0.005 0.005 0.02"/>
			</geometry>
			<material name="silver"/>
		</visual>
		<collision>
			<geometry>
				<box size="0.005 0.005 0.02"/>
			</geometry>
		</collision>
	</link>
	<joint name="r_g_base" type="revolute">
		<axis xyz="1 0 0"/>
		<limit velocity="1.0" effort="100.0" upper="0" lower="-0.5236"/>
		<origin rpy="0.9162 0 0" xyz="0 -0.02 0.035"/>
		<parent link="hand"/>
		<child link="right_gripper_base"/>
	</joint>
	<link name="left_gripper_finger">
		<inertial>
			<mass value="0.01"/>
			<inertia ixx="1.57e-7" iyy="1.57e-7" izz="6.8e-7" ixy="0" ixz="0" iyz="0"/>
		</inertial>
		<visual>
			<geometry>
				<box size="0.008 0.008 0.05"/>
			</geometry>
			<material name="silver"/>
		</visual>
		<collision>
			<geometry>
				<box size="0.008 0.008 0.05"/>
			</geometry>
		</collision>
	</link>
	<joint name="l_g_finger" type="fixed">
		<origin rpy="0.3926 0 0" xyz="0 -0.01 0.03"/>
		<parent link="left_gripper_base"/>
		<child link="left_gripper_finger"/>
	</joint>
	<link name="right_gripper_finger">
		<inertial>
			<mass value="0.01"/>
			<inertia ixx="1.57e-7" iyy="1.57e-7" izz="6.8e-7" ixy="0" ixz="0" iyz="0"/>
		</inertial>
		<visual>
			<geometry>
				<box size="0.008 0.008 0.05"/>
			</geometry>
			<material name="silver"/>
		</visual>
		<collision>
			<geometry>
				<box size="0.005 0.005 0.05"/>
			</geometry>
		</collision>
	</link>
	<joint name="r_g_finger" type="fixed">
		<origin rpy="-0.3926 0 0" xyz="0 0.01 0.03"/>
		<parent link="right_gripper_base"/>
		<child link="right_gripper_finger"/>
	</joint>

	<transmission name="tran0">
		<type>transmission_interface/SimpleTransmission</type>
		<joint name="hip">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
		</joint>
		<actuator name="motor0">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
			<mechanicalReduction>1</mechanicalReduction>
		</actuator>
	</transmission>
	<transmission name="tran1">
		<type>transmission_interface/SimpleTransmission</type>
		<joint name="shoulder">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
		</joint>
		<actuator name="motor1">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
			<mechanicalReduction>1</mechanicalReduction>
		</actuator>
	</transmission>
	<transmission name="tran2">
		<type>transmission_interface/SimpleTransmission</type>
		<joint name="elbow">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
		</joint>
		<actuator name="motor2">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
			<mechanicalReduction>1</mechanicalReduction>
		</actuator>
	</transmission>
	<transmission name="tran3">
		<type>transmission_interface/SimpleTransmission</type>
		<joint name="wrist">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
		</joint>
		<actuator name="motor3">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
			<mechanicalReduction>1</mechanicalReduction>
		</actuator>
	</transmission>
	<transmission name="tran4">
		<type>transmission_interface/SimpleTransmission</type>
		<joint name="l_g_base">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
		</joint>
		<actuator name="motor4">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
			<mechanicalReduction>1</mechanicalReduction>
		</actuator>
	</transmission>
	<transmission name="tran5">
		<type>transmission_interface/SimpleTransmission</type>
		<joint name="r_g_base">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
		</joint>
		<actuator name="motor5">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
			<mechanicalReduction>1</mechanicalReduction>
		</actuator>
	</transmission>

	<gazebo>
		<plugin name="control" filename="libgazebo_ros_control.so"/>
	</gazebo>
	<gazebo>
		<plugin name="joint_state_publisher" filename="libgazebo_ros_joint_state_publisher.so">
			<jointName>hip, shoulder, elbow, wrist</jointName>
		</plugin>
	</gazebo>
</robot>